sdk/search/Azure.Search.Documents/src/SearchServiceClient.cs

```diff
@@ -172,71 +184,227 @@ public virtual SearchIndexClient GetSearchIndexClient(string indexName)
+
+        /// <summary>
+        /// Gets an existing index from the Search service.
+        /// </summary>
+        /// <param name="indexName">
+        /// The name of the index.
+        /// </param>
+        /// <param name="options">
+        /// Optional <see cref="SearchRequestOptions"/> to customize the operation's behavior.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate notifications
+        /// that the operation should be canceled.
+        /// </param>
+        /// <returns>
+        /// The requested <see cref="SearchIndex"/>.
+        /// </returns>
+        /// <exception cref="ArgumentException">
+        /// <paramref name="indexName"/> is an empty string.
+        /// </exception>
+        /// <exception cref="ArgumentNullException">
+        /// <paramref name="indexName"/> is null.
+        /// </exception>
+        /// <exception cref="RequestFailedException">
+        /// Thrown when a failure is returned by the Search service.
+        /// </exception>
+        [ForwardsClientCalls]
+        public virtual Response<SearchIndex> GetIndex(
+            string indexName,
+            SearchRequestOptions options = null,
+            CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNullOrEmpty(indexName, nameof(indexName));
+
+            return IndexesClient.Get(
+                indexName,
+                options?.ClientRequestId,
+                cancellationToken);
+        }
+
+        /// <summary>
+        /// Gets an existing index from the Search service.
+        /// </summary>
+        /// <param name="indexName">
+        /// The name of the index.
+        /// </param>
+        /// <param name="options">
+        /// Optional <see cref="SearchRequestOptions"/> to customize the operation's behavior.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate notifications
+        /// that the operation should be canceled.
+        /// </param>
+        /// <returns>
+        /// The requested <see cref="SearchIndex"/>.
+        /// </returns>
+        /// <exception cref="ArgumentException">
+        /// <paramref name="indexName"/> is an empty string.
+        /// </exception>
+        /// <exception cref="ArgumentNullException">
+        /// <paramref name="indexName"/> is null.
+        /// </exception>
+        /// <exception cref="RequestFailedException">
+        /// Thrown when a failure is returned by the Search service.
+        /// </exception>
+        [ForwardsClientCalls]
+        public virtual async Task<Response<SearchIndex>> GetIndexAsync(
+            string indexName,
+            SearchRequestOptions options = null,
+            CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNullOrEmpty(indexName, nameof(indexName));
```

Comment:
If we tweak the swagger, will the generated code validate this for us?

Reply:
No. I've talked with @pakrym about this already. The only thing I could think of is if swagger put a `minLength` constaint on a string field of 1 and the generators support that; otherwise, it's up to earch service to validate string arguments for anything other than `null`.

Reply:
I don't mean tweak the code generator but rather make this a required field which the code generator should then validate for us.  Then we don't have to worry about consistency across language implementations, etc.  This is just a nit though and not something we need to block on.

Reply:
It is already a required field and the generator checks only for null; however, null is not the same as an empty string, which is valid unless a `minLength` is supplied. I have a separate conversation with @pakrym on this but this is necessary for consistency with the .NET BCL.